### PR TITLE
Add @Nullable annotation and requireNonNull

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/WorkProcessorUtils.java [hunk 1]
```diff
@@ -345,13 +345,13 @@ public T getResult()
 
     private static class ElementAndProcessor<T>
     {
-        final T element;
+        @Nullable final T element;
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/operator/WorkProcessorUtils.java]: Shouldn't this be on a separate (prev) line?

Comment (electrum) [reply presto-main/src/main/java/com/facebook/presto/operator/WorkProcessorUtils.java]: The code style doesn't enforce either way, but the style is to put it on a separate line. For example, `validationBuilder` field in `OrcWriter`.

Comment (electrum) [reply presto-main/src/main/java/com/facebook/presto/operator/WorkProcessorUtils.java]: BTW, I'd also make these fields private, so it's clear we don't intended to access them outside the class.

Comment (sopel39) [reply presto-main/src/main/java/com/facebook/presto/operator/WorkProcessorUtils.java]: > The code style doesn't enforce either way, but the style is to put it on a separate line. For example, validationBuilder field in OrcWriter.

The code style actually enforces it on a single line (without `private` keyword)

> BTW, I'd also make these fields private, so it's clear we don't intended to access them outside the class.

`private` keyword in nested class is not enforcing, therefore it's redundant here and possibly misleading. I prefer this convention (no `private` keyword in private nested classes fields/methods), but I understand that others can have different preference. I think that for such things we could have a code style guide (vide recent https://github.com/prestodb/presto/pull/11625#discussion_r222092062 discussion)

**Pull request merged**